client/restclient/restclient.go

```diff
@@ -235,8 +244,8 @@ func HandleResponse(response *http.Response, requestStr string) (responseData []
 		if err != nil {
 			return
 		}
-		uerr := json.Unmarshal(responseData, &errResp)
-		if uerr == nil {
+		myErr := json.Unmarshal(responseData, &errResp)
+		if myErr == nil {
 			err = fmt.Errorf("recieved error code %d: %s \n response: \n %s", response.StatusCode, requestStr, responseData)
```

Comment:
misspelling of "received"

Reply:
should this function also return an err here when `myErr` is non-nil?

Reply:
i had same thought when i looked into, but looks like idea of the HandleResponse function was to print corresponding errors based on http error code, so when there are parsing errors it is left to the caller to handle. in this above case we need to set err to myErr.

Reply:
it's not doing returning or handling `myErr` at all currently.

Reply:
yes i saw it, made corresponding change.